packages/client/src/publish/ProxyPublisher.ts

```diff
@@ -0,0 +1,30 @@
+import { StreamDefinition } from '../types'
+import { StreamIDBuilder } from '../StreamIDBuilder'
+import { inject, Lifecycle, scoped } from 'tsyringe'
+import BrubeckNode from '../BrubeckNode'
+import { Context } from '../utils/Context'
+import { ProxyDirection } from 'streamr-client-protocol'
+
+@scoped(Lifecycle.ContainerScoped)
+export default class ProxyPublisher {
+    constructor(
+        context: Context,
+        private node: BrubeckNode,
+        @inject(StreamIDBuilder) private streamIdBuilder: StreamIDBuilder,
+    ) {
+    }
+
+    async setPublishProxies(streamDefinition: StreamDefinition, nodeIds: string[]): Promise<void> {
```

Comment:
If user gives a `streamDefinition` without a partition, we will set proxies for all partitions of that stream? (Is that feature part of this PR, or maybe next PR?)